src/frontends/tensorflow_common/src/op/crop_and_resize.cpp

```diff
@@ -3,137 +3,93 @@
-                make_shared<Constant>(element::i64, Shape{4}, vector<int64_t>({1, stride_height, stride_width, 1}));
-
-            // crop
-            auto ng_crop_node =
-                make_shared<StridedSlice>(ng_input, begin, end, strides, vector<int64_t>{}, vector<int64_t>{});
-            auto ng_crop = ng_crop_node->output(0);
-
-            Interpolate::InterpolateAttrs interpolate_attrs;
-            // always corner aligned
-            interpolate_attrs.coordinate_transformation_mode = Interpolate::CoordinateTransformMode::ALIGN_CORNERS;
-
-            // TODO: handle the case when extrapolation value is greatger than 1.0
-            // arguments for resizing
-            auto ng_spatial_shape =
-                make_shared<Constant>(element::i32, Shape{2}, vector<int32_t>{crop_height, crop_width});
-            auto ng_input_shape = make_shared<Convert>(ng_spatial_shape, element::f32);
-            auto ng_crop_size = make_shared<Convert>(ng_size, element::f32);
-            auto ng_scales = make_shared<Divide>(ng_crop_size, ng_input_shape);
-            auto ng_axes = make_shared<Constant>(element::i32, Shape{2}, vector<int>({2, 3}));
-
-            if (resize_method == "bilinear") {
-                interpolate_attrs.mode = Interpolate::InterpolateMode::LINEAR;
-            } else {  // nearest
-                interpolate_attrs.mode = Interpolate::InterpolateMode::NEAREST;
-            }
-
-            ng_crop = make_transpose(ng_crop, {0, 3, 1, 2});
-            auto ng_output =
-                make_shared<Interpolate>(ng_crop, ng_size, ng_scales, ng_axes, interpolate_attrs)->output(0);
-            ng_output = make_transpose(ng_output, {0, 2, 3, 1});
-            ng_crop_outputs.at(i) = ng_output;
-        }
+OutputVector translate_crop_and_resize_bilinear(const NodeContext& node) {
+    default_op_checks(node, 4, {"CropAndResize"});
+    auto image = node.get_input(0);
+    auto boxes = node.get_input(1);
+    auto box_ind = node.get_input(2);
+    auto crop_size = node.get_input(3);
+
+    // retrieve attributes
+    auto method = node.get_attribute<string>("method", "bilinear");
+    auto extrapolation_value = node.get_attribute<float>("extrapolation_value", 0.0f);
+
+    // try to retrive crop_size as the constant, otherwise, we cannot support it
+    Shape crop_sizes;
+    get_const_input(node, 3, &crop_sizes);
+
+    // concatenate boxes and box_ind inputs because
+    // ROIPooling accepts ROIs in a format [batch_id, x_1, y_1, x_2, y_2]
+    // prepare box_ind for futher concatenation
+    auto const_one = make_shared<Constant>(element::i32, Shape{1}, 1);
+    box_ind = make_shared<Unsqueeze>(box_ind, const_one);
+    box_ind = make_shared<Convert>(box_ind, element::f32);
+    boxes = make_shared<Concat>(OutputVector{box_ind, boxes}, 1);
+
+    // boxes are going in the format [y1, x1, y2, x2]
+    // so we need to adjust them to the format [x_1, y_1, x_2, y_2]
+    // use Convolution for the swapping
```

Comment:
It looks like you can use `Gather` for that. `Convolution` is highly optimized, yes, but I doubt it can be better then `Gather`

Reply:
Done